slick/src/main/scala/slick/jdbc/JdbcDataSource.scala

```diff
@@ -173,6 +178,24 @@ object HikariCPJdbcDataSource extends JdbcDataSourceFactory {
     val ds = new HikariDataSource(hconf)
     new HikariCPJdbcDataSource(ds, hconf)
   }
+
+  def extractUrl(maybeUrl: Option[String]): (Option[String], Option[(String, String)]) = {
+    maybeUrl match {
+      case Some(PostgresFullUrl(username, password, host, dbname)) =>
+        Some(s"jdbc:postgresql://$host/$dbname") -> Some(username -> password)
+
+      case Some(url @ MysqlFullUrl(username, password, host, dbname)) =>
+        val defaultProperties = "?useUnicode=yes&characterEncoding=UTF-8&connectionCollation=utf8_general_ci"
```

Comment:
Yes. Apparently one of the changes that was made from Play 2.3 to 2.4 was to let Slick handle a lot more, including creating its own connection pool and reading the configuration.